### PR TITLE
Add relations recipe

### PR DESCRIPTION
**Pull request by koskimas**
First draft of the relations recipe. Only postgres jsonb examples for now.

### CODE REVIEW

File: site/docs/recipes/relations.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
+# Relations
+
+Kysely is just a query builder and doesn't have the concept of relations. Kysely builds the SQL you tell it to and nothing more. This doesn't mean there's no way to nest related rows in your queries. You just have to do it with the tools SQL and the dialect provide. In this recipe we show one way to do that on PostgreSQL.
```
Comment (igalklebanov) [inline site/docs/recipes/relations.md:3]:
```suggestion
<h3>
Kysely IS NOT an ORM. Kysely DOES NOT have the concept of relations.
Kysely IS a query builder. Kysely DOES build the SQL you tell it to, nothing more, nothing less.
</h3>

Phew, glad we got that out the way..

All that was said above doesn't mean there's no way to nest related rows in your queries. 
You just have to do it with the tools SQL and the underlying dialect (e.g. PostgreSQL) you're using provide. 
In this recipe we show one way to do that when using the PostgreSQL built-in dialect.
```

File: site/docs/recipes/relations.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+## PostgreSQL `jsonb` data type and functions
+
+PostgreSQL has a rich JSON support through its `json` and `jsonb` data types and functions. The node PostgreSQL driver `pg` automatically parses returned `json` and `jsonb` columns as objects. With the combination of these two things, we can write some super efficient queries with nested relations.
```
Comment (igalklebanov) [inline site/docs/recipes/relations.md:7]:
```suggestion
PostgreSQL has rich JSON support through its `json` and `jsonb` data types and functions. `pg`, the node PostgreSQL driver, automatically parses returned `json` and `jsonb` columns as json objects. With the combination of these two things, we can write some super efficient queries with nested relations.
```

File: site/docs/recipes/relations.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+Let's start with the raw SQL and then see how we can write the query using Kysely in a nice type-safe way.
```
Comment (igalklebanov) [inline site/docs/recipes/relations.md:9]:
```suggestion
Let's start with some raw SQL, and then see how we can write the query using Kysely in a nice type-safe way.
```

File: site/docs/recipes/relations.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+In the following query, we fetch a list of persons and for each person, we nest the person's pets and the person's mother into the returned objects:
```
Comment (igalklebanov) [inline site/docs/recipes/relations.md:11]:
```suggestion
In the following query, we fetch a list of people (from "person" table) and for each person, we nest the person's pets, and mother, into the returned objects:
```

File: site/docs/recipes/relations.md [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+        person as mother
+      WHERE 
+        mother.id = person.mother_id
+    ) mother
+  ) mother
+FROM
+  person
+```
+
+Simple right 😅. Yeah, not so much. But it does provide 100% control over the queries and a really good performance as long as you have indices for `pet.owner_id` and `person.mother_id`.
```
Comment (igalklebanov) [inline site/docs/recipes/relations.md:52]:
```suggestion
Simple right 😅. Yeah, not so much. But it does provide 100% control over the queries and a really good performance as long as you have indices (or indexes, we don't judge) for "pet.owner_id" and "person.mother_id".
```

File: site/docs/recipes/relations.md [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+
+Fortunately we can improve and simplify this a lot using kysely. First let's define couple of helpers:
```
Comment (igalklebanov) [inline site/docs/recipes/relations.md:54]:
```suggestion
Fortunately we can improve and simplify this a lot using Kysely. First let's define a couple of helpers:
```

File: site/docs/recipes/relations.md [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+}
+
+export function jsonObjectFrom<O>(
+  expr: Expression<O>
+): RawBuilder<Simplify<O>> {
+  return sql`(select to_jsonb(obj) from ${expr} as obj)`
+}
+```
+
+These helpers are included in kysely and you can import them from the `helpers` package like this:
```
Comment (igalklebanov) [inline site/docs/recipes/relations.md:70]:
```suggestion
These helpers are included in Kysely and you can import them from the `helpers` module like this:
```

**Review by igalklebanov — APPROVED** [review]
LGTM! Love the vibes in this one! 🤟

File: site/docs/recipes/relations.md (hunk 7 above)
Comment (koskimas) [reply site/docs/recipes/relations.md:70]: Too much golang 😅

File: site/docs/recipes/relations.md [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+You just have to do it with the tools SQL and the underlying dialect (e.g. PostgreSQL) you're using provide. 
+In this recipe we show one way to do that when using the PostgreSQL built-in dialect.
+
+## PostgreSQL `jsonb` data type and functions
+
+PostgreSQL has rich JSON support through its `json` and `jsonb` data types and functions. `pg`, the node PostgreSQL driver, automatically parses returned `json` and `jsonb` columns as json objects. With the combination of these two things, we can write some super efficient queries with nested relations.
+
+Let's start with some raw SQL, and then see how we can write the query using Kysely in a nice type-safe way.
+
+In the following query, we fetch a list of people (from "person" table) and for each person, we nest the person's pets, and mother, into the returned objects:
```
Comment (fhur) [inline site/docs/recipes/relations.md:20]:
```suggestion
In the following query, we fetch a list of people (from "person" table) and for each person, we nest the person's pets and owner, into the returned objects:
```

File: site/docs/recipes/relations.md [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+
+```ts
```
Comment (fhur) [inline site/docs/recipes/relations.md:22]:
```suggestion
```sql
```

Comment (koskimas) [reply site/docs/recipes/relations.md:22]: Oof, nice catch!

File: site/docs/recipes/relations.md [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+      FROM
+        pet
+      WHERE 
+        pet.owner_id = person.id
+      ORDER BY 
+        pet.name
+    ) pets
+  ) pets,
+
+  // Select person's mother as a jsonb object
```
Comment (tonivj5) [inline site/docs/recipes/relations.md:43]:
```suggestion
  # Select person's mother as a jsonb object
```

**Review by tonivj5 — COMMENTED** [review]
using sql, use `#` as comments to avoid this
image

File: site/docs/recipes/relations.md (hunk 10 above)
Comment (koskimas) [reply site/docs/recipes/relations.md:43]: Good catch, actually should be `--`

**Pull request merged**